include/deal.II/base/exceptions.h

```diff
@@ -1512,9 +1512,10 @@ namespace deal_II_exceptions
 #ifdef DEAL_II_HAVE_BUILTIN_EXPECT
 #  define AssertThrow(cond, exc)                                       \
     {                                                                  \
-      if (__builtin_expect(!(cond), false))                            \
+      if (!(cond))                                                     \
         ::dealii::deal_II_exceptions::internals::issue_error_noreturn( \
-          ::dealii::deal_II_exceptions::internals::throw_on_exception, \
+          ::dealii::deal_II_exceptions::internals::                    \
+            abort_or_throw_on_exception,                               \
```

Comment:
This turns `AssertThrow` into the exact same thing as `Assert` if I see correctly (except that it is always enabled in `Release` mode). I don't think that we want this. Blocking.